pkg/util/modules/modules_test.go

```diff
@@ -47,3 +47,109 @@ func TestDependencies(t *testing.T) {
+	sut := NewManager()
+	sut.RegisterModule("module1", mockInitFunc)
+
+	m := sut.modules["module1"]
+
+	assert.NotNil(t, mockInitFunc, m.initFn, "initFn not assigned")
+	assert.True(t, m.public, "module should be public")
+}
+
+func TestRegisterModuleSetsDefaultOption(t *testing.T) {
+	sut := NewManager()
+	sut.RegisterModule("module1", mockInitFunc)
+
+	m := sut.modules["module1"]
+
+	assert.True(t, m.public, "mould should be public")
+}
+
+func TestFunctionalOptAtTheEndWins(t *testing.T) {
+	publicMod := func(option *module) {
+		option.public = true
+	}
+	sut := NewManager()
+	sut.RegisterModule("mod1", mockInitFunc, PrivateModule, publicMod, PrivateModule)
+
+	m := sut.modules["mod1"]
+
+	assert.NotNil(t, mockInitFunc, m.initFn, "initFn not assigned")
+	assert.False(t, m.public, "module should be public")
+}
+
+func TestGetAllPublicModulesNames(t *testing.T) {
+	sut := NewManager()
+	sut.RegisterModule("public1", mockInitFunc)
+	sut.RegisterModule("public2", mockInitFunc)
+	sut.RegisterModule("public3", mockInitFunc)
+	sut.RegisterModule("private1", mockInitFunc, PrivateModule)
+	sut.RegisterModule("private2", mockInitFunc, PrivateModule)
+
+	pm := sut.PublicModuleNames()
+
+	assert.Len(t, pm, 3, "wrong result slice size")
+	assert.Contains(t, pm, "public1", "missing public module")
+	assert.Contains(t, pm, "public2", "missing public module")
+	assert.Contains(t, pm, "public3", "missing public module")
+}
+
+func TestGetAllPublicModulesNamesHasNoDupWithDependency(t *testing.T) {
+	sut := NewManager()
+	sut.RegisterModule("public1", mockInitFunc)
+	sut.RegisterModule("public2", mockInitFunc)
+	sut.RegisterModule("public3", mockInitFunc)
+
+	assert.NoError(t, sut.AddDependency("public1", "public2", "public3"))
+
+	pm := sut.PublicModuleNames()
+
+	// make sure we don't include any module twice because there is a dependency
+	assert.Len(t, pm, 3, "wrong result slice size")
+	assert.Contains(t, pm, "public1", "missing public module")
+	assert.Contains(t, pm, "public2", "missing public module")
+	assert.Contains(t, pm, "public3", "missing public module")
+}
+
+func TestGetEmptyListWhenThereIsNoPublicModule(t *testing.T) {
+	sut := NewManager()
+	sut.RegisterModule("private1", mockInitFunc, PrivateModule)
+	sut.RegisterModule("private2", mockInitFunc, PrivateModule)
+	sut.RegisterModule("private3", mockInitFunc, PrivateModule)
+	sut.RegisterModule("private4", mockInitFunc, PrivateModule)
+
+	pm := sut.PublicModuleNames()
+
+	assert.Len(t, pm, 0, "wrong result slice size")
+}
+
+func TestPublicModuleNamesReturnsSortedList(t *testing.T) {
+	sut := NewManager()
+	sut.RegisterModule("c", mockInitFunc)
+	sut.RegisterModule("b", mockInitFunc)
+	sut.RegisterModule("a", mockInitFunc)
+
+	pm := sut.PublicModuleNames()
+
+	assert.Len(t, pm, 3, "wrong result slice size")
+	assert.Equal(t, []string{"a", "b", "c"}, pm, "module names list is not sorted in ascending order")
+}
+
+func TestIsPublicModule(t *testing.T) {
```

Comment:
It feels like this could easily be merged into one of the previous tests (`TestRegisterModuleWithOptions` or `TestRegisterModuleSetsDefaultOption`). But that's just nit-picking.

Reply:
I would think that having separate test for `IsPublicModule` is better because it adheres to single responsibility principle more. Also, when this test fails, it's much easer to tell which method is at fault.

Reply:
When tests fail, we get a line number where they failed, so it's easy to find out. If you want to keep the test, I'm fine with that. It just felt like lot of repeated boilerplate to me.